src/P2P/Node/PeerDB.hs

```diff
@@ -258,14 +276,13 @@ addPeerEntry b m = m & case getOne (getEQ addr m) of
   where
     addr = _peerAddr $ _peerEntryInfo b
     replace = updateIx addr b
-    update a = updateIx addr $!! PeerEntry
-        { _peerEntryInfo = _peerEntryInfo a
-        , _peerEntrySuccessiveFailures = _peerEntrySuccessiveFailures a + _peerEntrySuccessiveFailures b
-        , _peerEntryLastSuccess = max (_peerEntryLastSuccess a) (_peerEntryLastSuccess b)
-        , _peerEntryNetworkIds = _peerEntryNetworkIds a <> _peerEntryNetworkIds b
-        , _peerEntryActiveSessionCount = _peerEntryActiveSessionCount a + _peerEntryActiveSessionCount b
-        , _peerEntrySticky = PeerEntrySticky False
-        }
+    -- we retain the old mark for the peer
+    update a = updateIx addr $! a
+            & peerEntrySuccessiveFailures +~ _peerEntrySuccessiveFailures b
+            & peerEntryLastSuccess %~ max (_peerEntryLastSuccess b)
+            & peerEntryNetworkIds <>~ _peerEntryNetworkIds b
+            & peerEntryActiveSessionCount +~ _peerEntryActiveSessionCount b
```

Comment:
did you double check that this is actually as strict as the original code? The `PeerEntry` should be NF.

(I always find it hard to reason about strictness when lenses are involved)